### PR TITLE
Feature/28 local base urls

### LINKED ISSUE #28: Add explicit setting to determine if url is local
If several services use the same gateaway or hosted on the same domain it could be tricky to determine if url is local or external.
We can set up a setting `LOOSE_FK_LOCAL_BASE_URLS` which will include a list of prefixes (hostname + subpath) which determine local urls
The setting should be optional

Related issue - https://github.com/open-zaak/open-zaak/issues/1592

### PR DESCRIPTION
**Pull request by annashamray**
fixes #28

### CODE REVIEW

File: django_loose_fk/lookups.py [hunk 1]
```diff
@@ -116,6 +116,14 @@ def get_prep_lookup(self):
         if self.rhs_is_direct_value():
             # If we get here, we are dealing with single-column relations.
             self.rhs = [get_normalized_value(val)[0] for val in self.rhs]
+
+        else:
```
Comment (alextreme) [inline django_loose_fk/lookups.py:120]: @annashamray is this change necessary? It doesn't feel relevant (but it could be my lack of knowledge in this library)

File: django_loose_fk/utils.py [hunk 2]
```diff
@@ -7,6 +9,21 @@
 from rest_framework.request import Request
 
 
+def is_local(host: str, url: str) -> bool:
```
Comment (alextreme) [inline django_loose_fk/utils.py:12]: Could we have a few unittests for only this method? I'm all in favor of integration tests but this feels complicated enough to test separately as well

File: django_loose_fk/lookups.py (hunk 1 above)
Comment (annashamray) [reply django_loose_fk/lookups.py:120]: We are doing some shady stuff with custom lookups here, and it looks like django method used here was changed, so tests are failing without it - https://github.com/maykinmedia/django-loose-fk/actions/runs/8552960317/job/23435156986
So I had to change it to pass CI.

**Review by alextreme — APPROVED** [review]
Request for additional tests, but imho this isn't mandatory. A double-check from Steven is though

File: django_loose_fk/loaders.py [hunk 3]
```diff
@@ -54,6 +55,10 @@ def is_local_url(self, url: str) -> bool:
                 "break django-loose-fk's behaviour. You should use an explicit list.",
                 RuntimeWarning,
             )
+            return validate_host(parsed.netloc, allowed_hosts)
```
Comment (stevenbal) [inline django_loose_fk/loaders.py:58]: I was wondering if there is a particular reason we don't do `validate_host` in the `is_local` util itself? I see that in `filters.py` and `drf.py` `is_local` is used without validating if it matches with `ALLOWED_HOSTS`

**Review by stevenbal — COMMENTED** [review]
Just a question regarding `validate_host`, other than that it looks good. I would also like to see the unittests that Alex mentioned, but this could maybe be added in a separate PR if we want to get this released quickly

File: django_loose_fk/loaders.py (hunk 3 above)
Comment (annashamray) [reply django_loose_fk/loaders.py:58]: It was not developed by me, but I have a theory:) in the Resolver we have `host` as parameter and it's used where we can use `host` from request. Here we don't know `host`, therefore we compare it with ALLOWED_HOSTS

Comment (stevenbal) [reply django_loose_fk/loaders.py:58]: Ah that makes sense :+1:

**Review by stevenbal — APPROVED** [review]

File: django_loose_fk/utils.py (hunk 2 above)
Comment (annashamray) [reply django_loose_fk/utils.py:12]: Tests are added
